lib/active_merchant/billing/gateways/deepstack.rb

```diff
@@ -0,0 +1,388 @@
+        post[:meta][:shipping_info][:address] = {}
+        post[:meta][:shipping_info][:address][:line_1] = shipping[:address1] if shipping[:address1]
+        post[:meta][:shipping_info][:address][:line_2] = shipping[:address2] if shipping[:address2]
+        post[:meta][:shipping_info][:address][:city] = shipping[:city] if shipping[:city]
+        post[:meta][:shipping_info][:address][:state] = shipping[:state] if shipping[:state]
+        post[:meta][:shipping_info][:address][:postal_code] = shipping[:zip] if shipping[:zip]
+        post[:meta][:shipping_info][:address][:country_code] = shipping[:country] if shipping[:country]
+      end
+
+      def add_invoice(post, money, authorization, options)
+        post[:amount] = amount(money)
+        post[:charge] = authorization
+        # post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_payment(post, payment, options)
+        if payment.kind_of?(String)
+          post[:source] = {}
+          post[:source][:type] = 'card_on_file'
+          post[:source][:card_on_file] = {}
+          post[:source][:card_on_file][:id] = payment
+          post[:source][:card_on_file][:cvv] = options[:verification_value] || '' # Should we fail if no cvv?
+          post[:source][:card_on_file][:customer_id] = options[:customer_id] || ''
+        # credit card object
+        elsif payment.respond_to?(:number)
+          post[:source] = {}
+          post[:source][:type] = 'credit_card'
+          post[:source][:credit_card] = {}
+          post[:source][:credit_card][:account_number] = payment.number
+          post[:source][:credit_card][:cvv] = payment.verification_value || '' # Should we fail if no cvv?
+          post[:source][:credit_card][:expiration] = '%02d%02d' % [payment.month, payment.year % 100]
+          post[:source][:credit_card][:customer_id] = options[:customer_id] || ''
+        end
+      end
+
+      def add_payment_instrument(post, creditcard, options)
+        if creditcard.kind_of?(String)
+          post[:source] = creditcard
+          return post
+        end
+        return post unless creditcard.respond_to?(:number)
+
+        post[:payment_instrument] = {}
+        post[:payment_instrument][:type] = 'credit_card'
+        post[:payment_instrument][:credit_card] = {}
+        post[:payment_instrument][:credit_card][:account_number] = creditcard.number
+        post[:payment_instrument][:credit_card][:expiration] = '%02d%02d' % [creditcard.month, creditcard.year % 100]
+        post[:payment_instrument][:credit_card][:cvv] = creditcard.verification_value
+      end
+
+      def add_order(post, amount, options)
+        post[:transaction] = {} unless post.key?(:transaction)
+
+        post[:transaction][:amount] = amount
+        post[:transaction][:cof_type] = options.key?(:cof_type) ? options[:cof_type].upcase : 'UNSCHEDULED_CARDHOLDER'
+        post[:transaction][:capture] = false # Change this in the request (auth/charge)
```

Comment:
Setting `post[:transaction][:capture]` to `false` here with the expectation that it's set to `true` elsewhere makes the value of this field dependent on the order in which `add_order` is called, which could lead to the accidental introduction of bugs in the future. Do you have to set a value for `post[:transaction][:capture]` here at all? Is the absence of the field also treated as `false` by the API? 

If you do need to set this field to `false`, `post[:transaction][:capture] ||= false` would guard against overriding a previously set truthy value.

Reply:
Hello, `post[:transaction][:capture]` is required by the API and I would like to keep it defaulted to false until it is set true by `purchase`